mne/io/nsx/nsx.py

```diff
@@ -0,0 +1,601 @@
+
+    ver_major, ver_minor = basic_header.pop("ver_major"), basic_header.pop("ver_minor")
+    basic_header["spec"] = "{}.{}".format(ver_major, ver_minor)
+
+    data_header = list()
+    index = 0
+    offset = basic_header["bytes_in_headers"]
+    filesize = _get_file_size(fname)
+    if float(basic_header["spec"]) < 3.0:
+        dtype2 = nsx_header_dict["data>2.1<3"]
+    else:
+        dtype2 = nsx_header_dict["data>=3"]
+    while offset < filesize:
+        dh = np.memmap(fname, dtype=dtype2, shape=1, offset=offset, mode="r")[0]
+        data_header.append(
+            {
+                "header": dh["header"],
+                "timestamp": dh["timestamp"],
+                "nb_data_points": dh["nb_data_points"],
+                "offset_to_data_block": offset + dh.dtype.itemsize,
+            }
+        )
+        # data size = number of data points * (data_bytes * number of channels)
+        # use of `int` avoids overflow problem
+        data_size = (
+            int(dh["nb_data_points"])
+            * int(basic_header["channel_count"])
+            * DATA_BYTE_SIZE
+        )
+        # define new offset (to possible next data block)
+        offset = data_header[index]["offset_to_data_block"] + data_size
+        index += 1
+
+    basic_header["data_header"] = data_header
+    return basic_header
+
+
+def _get_hdr_info(fname, stim_channel=True, eog=None, misc=None):
+    """Read header information NSx file."""
+    eog = eog if eog is not None else []
+    misc = misc if misc is not None else []
+
+    nsx_info = _read_header(fname)
+    ch_names = list(nsx_info["extended"]["electrode_label"])
+    ch_types = list(nsx_info["extended"]["type"])
+    ch_units = list(nsx_info["extended"]["units"])
+    ch_names, ch_types, ch_units = [
+        list(map(bytes.decode, xx)) for xx in (ch_names, ch_types, ch_units)
+    ]
+    max_analog_val = nsx_info["extended"]["max_analog_val"].astype("double")
+    min_analog_val = nsx_info["extended"]["min_analog_val"].astype("double")
+    max_digital_val = nsx_info["extended"]["max_digital_val"].astype("double")
+    min_digital_val = nsx_info["extended"]["min_digital_val"].astype("double")
+    cals = (max_analog_val - min_analog_val) / (max_digital_val - min_digital_val)
+
+    stim_channel_idxs, _ = _check_stim_channel(stim_channel, ch_names)
+
+    nchan = nsx_info["channel_count"]
+    logger.info("Setting channel info structure...")
+    chs = list()
+    pick_mask = np.ones(len(ch_names))
+
+    chs_without_types = list()
+    orig_units = {}
+
+    for idx, ch_name in enumerate(ch_names):
+        chan_info = {}
+        chan_info["logno"] = int(nsx_info["extended"]["electrode_id"][idx])
+        chan_info["scanno"] = int(nsx_info["extended"]["electrode_id"][idx])
+        chan_info["ch_name"] = ch_name
+        ch_unit = ch_units[idx]
+        if ch_unit == "":
+            chan_info["unit_mul"] = FIFF.FIFF_UNITM_NONE
+            chan_info["unit"] = FIFF.FIFF_UNIT_UNITLESS
```

Comment:
I think `UNIT_NONE` would be more appropriate if it's unknown unless it actually is a unitless quantity (like a ratio)